crowbar_framework/app/controllers/installer_controller.rb

```diff
@@ -0,0 +1,135 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+class InstallerController < ApplicationController
+  skip_before_filter :crowbar_mode
+  before_filter :hide_navigation
+
+  def index
+    @steps = all_steps
+  end
+
+  #
+  # Perform Crowbar Installation
+  #
+  # Provides the restful api call for
+  # /installer/status 	GET 	return done steps and error
+  # returns a hash with an indicator if the installation failed and the steps that are done
+  def status
+    if successful?
+      ENV["CROWBAR_MODE"] = nil if ENV["CROWBAR_MODE"]
+      respond_to do |format|
+        format.json { redirect_to root_path }
+        format.html { redirect_to root_path }
+      end
+    else
+      respond_to do |format|
+        format.json { render json: steps_done }
+        format.html { redirect_to installer_url }
+      end
+    end
+  end
+
+  #
+  # Set the status of an installation step
+  #
+  # Provides the restful api call for
+  # /installer/set_status 	POST 	set the status
+  # touch a dotfile in /opt/dell named after the step
+  # not really needed yet, but maybe later
+  def set_status
+    step = params[:step]
+    return render_not_found unless all_steps.include? step.to_sym
+    ret, _msg = write_file(step)
+    respond_to do |format|
+      case ret
+      when 200
+        format.json { head :ok }
+        format.html { redirect_to installer_url }
+      when 404
+        render_not_found
+      end
+    end
+  end
+
+  #
+  # Perform Crowbar Installation
+  #
+  # Provides the restful api call for
+  # /installer/install 	POST 	triggers install-chef-suse.sh
+  # the bash Process will be spawned in the background and therefore has not a direct return value which we can use here
+  def install
+    crowbar_dir = Rails.root.join("..")
+    install_ok = Rails.root.join(".crowbar-installed-ok")
+    if File.exist?(install_ok)
+      FileUtils.rm(install_ok)
+    end
+
+    unless File.exist?("#{crowbar_dir}/.crowbar_installing")
+      pid = spawn("sudo #{crowbar_dir}/bin/install-chef-suse.sh --crowbar")
+      write_file(crowbar_dir, ".crowbar_installing")
+      Process.detach(pid)
+    end
+
+    respond_to do |format|
+      format.json { head :ok }
+      format.html { redirect_to installer_url }
+    end
+  end
+
+  protected
+
+  def hide_navigation
+    @hide_navigation = true
+  end
+
+  def all_steps
+    [
+      :pre_sanity_checks,
+      :run_services,
+      :initial_chef_client,
+      :barclamp_install,
+      :bootstrap_crowbar_setup,
+      :apply_crowbar_config,
+      :transition_crowbar,
+      :chef_client_daemon,
+      :post_sanity_checks
+    ]
+  end
+
+  def steps_done
+    steps = all_steps.select do |step|
+      File.exist?(Rails.root.join("..", ".#{step}"))
+    end
+    { steps: steps, failed: failed? }
+  end
+
+  def failed?
+    File.exist?(Rails.root.join(".crowbar-install-failed"))
```

Comment:
Oh come on, please read the docs about Pathname... `Rails.root.join(".crowbar-install-failed").exist?`